l10n_ch_fds_upload_sepa/wizards/fds_postfinance_file_pain000.py

```diff
@@ -0,0 +1,48 @@
+from odoo import models, api
+
+import base64
+import logging
+
+_logger = logging.getLogger(__name__)
+
+
+class FdsPostfinanceFilePain000(models.Model):
+    _inherit = 'fds.postfinance.file'
+
+    @api.multi
+    def import2bankStatements(self):
+
+        account_pain000 = self.env['account.pain000.parser']
+        for pf_file in self:
+            try:
+                decoded_file = base64.b64decode(pf_file.data)
+
+                result = account_pain000.parse(decoded_file)
+
+                if result[0]:
+                    if result[1] is not None and result[1].id:
+                        # Link the payment order to the file import.
+                        pf_file.payment_order = result[1].id
+                        # Attach the file to the payment order.
+                        self.env['ir.attachment'].create({
+                                'datas_fname': pf_file.filename,
+                                'res_model': 'account.payment.order',
+                                'datas': pf_file.data,
+                                'name': pf_file.filename,
+                                'res_id': result[1].id})
+
+                    pf_file.write({
+                        'state': 'done',
+                        'data': pf_file.data
+                    })
+
+                    _logger.info("[OK] import file '%s' as pain.000",
+                                 pf_file.filename)
+
+                    self -= pf_file
```

Comment:
Never change the variable you are looping in. It will produce strange results. You should have taken the example I gave you.